Sources/PrimerSDK/Classes/Core/BIN Data/CardValidationService.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  CardValidationService.swift
+//  PrimerSDK
+//
+//  Created by Jack Newcombe on 30/10/2023.
+//
+
+import Foundation
+
+protocol CardValidationService {
+    func validateCardNetworks(withCardNumber cardNumber: String)
+}
+
+class DefaultCardValidationService: CardValidationService, LogReporter {
+    
+    var delegate: PrimerHeadlessUniversalCheckoutRawDataManagerDelegate? {
+        return self.rawDataManager.delegate
+    }
+        
+    let apiClient: PrimerAPIClientBINDataProtocol
+    
+    let debouncer: Debouncer
+    
+    let rawDataManager: PrimerHeadlessUniversalCheckout.RawDataManager
+
+    var mostRecentCardNumber: String?
+
+    init(rawDataManager: PrimerHeadlessUniversalCheckout.RawDataManager,
+         apiClient: PrimerAPIClientBINDataProtocol = PrimerAPIClient(),
+         debouncer: Debouncer = .init(delay: 0.35)) {
+        self.rawDataManager = rawDataManager
+        self.apiClient = apiClient
+        self.debouncer = debouncer
+    }
+        
+    func validateCardNetworks(withCardNumber cardNumber: String) {
+        let sanitizedCardNumber = cardNumber.replacingOccurrences(of: " ", with: "")
+        guard !sanitizedCardNumber.isEmpty, sanitizedCardNumber != mostRecentCardNumber else {
+            return
+        }
+        mostRecentCardNumber = sanitizedCardNumber
+        
+        let cardState = PrimerCardState(cardNumber: sanitizedCardNumber)
+        guard sanitizedCardNumber.count >= 6 else {
+            useLocalValidation(withCardState: cardState)
+            return
+        }
+        
+        debouncer.debounce { [weak self] in
+            self?.useRemoteValidation(withCardState: cardState)
+        }
+    }
+    
+    private func useRemoteValidation(withCardState cardState: PrimerCardState) {
+        delegate?.primerRawDataManager?(rawDataManager,
+                                        willFetchCardMetadataForState: cardState)
+        
+        let rawDataManager = rawDataManager
+        
+        _ = listCardNetworks(cardState.cardNumber).done { [weak self] result in
+            guard result.networks.count > 0 else {
+                self?.useLocalValidation(withCardState: cardState)
+                return
+            }
+            let cardMetadata = PrimerCardMetadata(availableCardNetworks: result.networks.map { network in
+                PrimerCardNetwork(displayName: network.displayName, networkIdentifier: network.value)
+            })
+            
+            self?.delegate?.primerRawDataManager?(rawDataManager,
+                                                  didReceiveCardMetadata: cardMetadata,
+                                                  forCardState: cardState)
+        }.catch { error in
+            // JN TODO: CHKT-1772 - send event
```

Comment:
I usually write it like this:
// TODO: (BNI) comment

BNI is for (B)oris (NI)kolić of course

This way you have this feature enabled :) 
<img width="1381" alt="Screenshot 2023-11-07 at 08 02 04" src="https://github.com/primer-io/primer-sdk-ios/assets/137198782/73e588b1-d582-4a9d-91f9-e2410090bd2b">
 you save

Reply:
Ah nice, I have a saved search in Xcode to make sure I sort them all out before they get to master